File: DSA With C/Using Array/priority_queue.c
//wap to implement priority queue using array
#include<stdio.h>
#include<stdlib.h>
#define MAX 4
void insert(int);
void del(int);
void check(int);
void display();
int pri_que[MAX], f = -1, r = -1;
void main()
{
    int n, ch;
    printf("\n1 - Insert an element into queue\n2 - Delete an element from queue\n3 - Exit");
    while (1)
    {
        printf("\nEnter your choice : ");    
        scanf("%d", &ch);
        switch (ch)
        {
            case 1: 
                printf("\nEnter value to be inserted : ");
                scanf("%d",&n);//n=15 10
                insert(n);//calling 
                break;
            case 2:
                printf("\nEnter value to delete : ");
                scanf("%d",&n);//n=50
                del(n);//calling 
                break;
            case 3: 
                exit(0);
            default: 
                printf("\nChoice is incorrect, Enter a correct choice");
        }
    }
}
void insert(int data)
{
    if (r >= MAX - 1)
    {
        printf("\nQueue overflow no more elements can be inserted");
        return;
    }
    if ((f == -1) && (r == -1))
    {
        f++;
        pri_que[++r] = data;
        return;
    }    
    else
        check(data);//calling 
    r++;//1
    display();
}
void check(int data)
{
    int i,j;
    for ( i = 0; i <= r; i++)
    {
        if (data >= pri_que[i])//10>=15
        {
            for ( j = r + 1; j > i; j--)
                pri_que[j] = pri_que[j - 1];
            pri_que[i] = data;
            return;
        }
    }
    pri_que[i] = data;//
}
void del(int data)
{
    int i;
    if ((f==-1) && (r==-1))
    {
        printf("\nQueue is empty no elements to delete");
        return;
    }
    for (i = 0; i <= r; i++)
    {
        if (data == pri_que[i])
        {
            for (; i < r; i++)//i=0  0<0 i++
                pri_que[i] = pri_que[i + 1];
            pri_que[i] = 0;
            r--;//-1
            if (r == -1) 
                f = -1;
            return;
        }
    }
    printf("\n%d not found in queue to delete", data);
    display();
}
void display()
{
    if ((f == -1) && (r == -1))
    {
        printf("\nQueue is empty");
        return;
    }
    for (int i=f; i<= r; i++)
        printf(" %d ", pri_que[i]);
}

File: DSA With Java/Code/using Array/Priority_queue.java
import java.util.Scanner;
class Priority_queue
{
    static final int MAX = 4;
    static int[] priQue = new int[MAX];
    static int front = -1, rear = -1;
    static Scanner sc = new Scanner(System.in);
    public static void main(String[] args) {
        int n, ch;
        System.out.println("1 - Insert an element into queue");
        System.out.println("2 - Delete an element from queue");
        System.out.println("3 - Exit");

        while (true) {
            System.out.print("\nEnter your choice : ");
            ch = sc.nextInt();
            switch (ch) {

                case 1:
                    insertByPriority();
                    break;
                case 2:
                    System.out.print("Enter value to delete : ");
                    n = sc.nextInt();
                    deleteByPriority(n);
                    break;
                case 3:
                    System.exit(0);
                default:
                    System.out.println("Choice is incorrect, Enter a correct choice");

            }

        }

    }

    static void insertByPriority() {
        if (rear >= MAX - 1) {
            System.out.println("Queue overflow, no more elements can be inserted");
            return;
        }
         System.out.print("Enter value to be inserted : ");
         int n = sc.nextInt();
        if (front == -1 && rear == -1) {
            front++;
            rear++;
            priQue[rear] = n;
            return;

        } else {
            check(n);
            rear++;

        }
        displayPQueue();
    }

    static void check(int data) {
        int i, j;
        for (i = 0; i <= rear; i++) {
            if (data >= priQue[i]) {
                for (j = rear + 1; j > i; j--) {
                    priQue[j] = priQue[j - 1];
                }
                priQue[i] = data;
                return;

            }

        }
        priQue[i] = data;
    }

    static void deleteByPriority(int data) {
      if (front == -1 && rear == -1) {
            System.out.println("Queue is empty, no elements to delete");
            return;}
      else{
        int f=0;
        for (int i = 0; i <= rear; i++) {
            if (data == priQue[i]) {
                for (; i < rear; i++) {
                    priQue[i] = priQue[i + 1];
                }
                priQue[rear] = 0;
                rear--;
                if (rear == -1)
                    front = -1;
                f++;
            } 
        }
        System.out.println((f==0)?data+" not found in queue to delete":"\nDeleting............\n");
    }
    displayPQueue();
    }

    static void displayPQueue() {
        if (front == -1 && rear == -1) {
            System.out.println("Queue is empty");
            return;
        }
        System.out.print("Queue elements: ");
        for (int i = front; i <= rear; i++) {
            System.out.print(priQue[i] + " ");
        }
        System.out.println();
    }
}